src/lib/transition_handler/breadcrumb_builder.ml

```diff
@@ -92,63 +92,68 @@ let build_subtrees_of_breadcrumbs ~logger ~precomputed_values ~verifier
                              to current external transition's parent hash"))
                 in
                 let open Deferred.Let_syntax in
-                match%bind
-                  O1trace.trace_recurring "Breadcrumb.build" (fun () ->
-                      Transition_frontier.Breadcrumb.build ~logger
-                        ~precomputed_values ~verifier ~trust_system ~parent
-                        ~transition:mostly_validated_transition
-                        ~sender:(Some sender) () )
-                with
-                | Ok new_breadcrumb ->
-                    let open Result.Let_syntax in
-                    Coda_metrics.(
-                      Counter.inc_one
-                        Transition_frontier_controller
-                        .breadcrumbs_built_by_builder) ;
-                    Deferred.return
-                      (let%map (_ : Transition_frontier.Breadcrumb.t) =
-                         breadcrumb_if_present
-                           (Logger.extend logger
-                              [("Check", `String "After creating breadcrumb")])
-                       in
-                       new_breadcrumb)
-                | Error err -> (
-                    (* propagate bans through subtree *)
-                    let subtree_nodes = Rose_tree.flatten subtree in
-                    let ip_address_set =
-                      let sender_from_tree_node node =
-                        Envelope.Incoming.sender (Cached.peek node)
+                try
```

Comment:
This `try` may not work if the async scheduler decides to run the code in a different thread. You should change this to use `Deferred.Or_error.try_with` or similar.

Reply:
Ok. I would do that.